tractolearn/tractoio/utils.py

```diff
@@ -0,0 +1,336 @@
+
+
+def load_streamlines(
+    fname: str,
+    ref_anat_fname: str,
+    streamlines_class: int,
+    resample: bool = False,
+    num_points: int = 256,
+    flip_all_streamlines: bool = False,
+) -> Tuple[np.array, np.array]:
+
+    img = load_ref_anat_image(ref_anat_fname)
+
+    # Read streamlines
+    to_space = Space.RASMM
+    trk_header_check = False
+    bbox_valid_check = False
+
+    tractogram = load_tractogram(
+        fname,
+        img.header,
+        to_space=to_space,
+        trk_header_check=trk_header_check,
+        bbox_valid_check=bbox_valid_check,
+    )
+
+    streamlines = tractogram.streamlines
+
+    if len(streamlines) == 0:
+        raise RuntimeError(
+            "Tractogram in filename {} contains no "
+            "streamlines. Please remove the file from the "
+            "experiment.".format(fname)
+        )
+
+    if resample:
+        streamlines = resample_streamlines(streamlines, num_points, arc_length=True)
+
+    if flip_all_streamlines:
+        streamlines = flip_streamlines(streamlines)
+
+    # Dump streamline data to array
+    streamlines_data = np.vstack(
+        [
+            streamlines[i][
+                np.newaxis,
+            ]
+            for i in range(len(streamlines))
+        ]
+    )
+
+    streamlines_classes = np.hstack(
+        [np.repeat(streamlines_class, len(streamlines_data))]
+    )
+
+    return streamlines_data, streamlines_classes
+
+
+@functools.lru_cache(maxsize=2)
+def load_ref_anat_image(ref_anat_fname):
+
+    img = nib.load(ref_anat_fname)
+    return img
+
+
+def load_process_streamlines2(
+    fname,
+    ref_anat_fname,
+    streamline_class_name,
+    random_flip=True,
+    random_flip_ratio=0.3,
+):
+
+    img = load_ref_anat_image(ref_anat_fname)
+
+    # Read streamlines
+    to_space = Space.RASMM
+    trk_header_check = False
+    bbox_valid_check = False
+    tractogram = load_tractogram(
+        fname,
+        img.header,
+        to_space=to_space,
+        trk_header_check=trk_header_check,
+        bbox_valid_check=bbox_valid_check,
+    )
+
+    if len(tractogram.streamlines) == 0:
+        raise RuntimeError(
+            "Tractogram in filename {} contains no "
+            "streamlines. Please remove the file from the "
+            "experiment.".format(fname)
+        )
+
+    # Apply random flip
+    if random_flip:
+        flipped_streamlines, _ = flip_random_streamlines(
+            tractogram.streamlines, random_flip_ratio
+        )
+    else:
+        flipped_streamlines = tractogram.streamlines
+
+    # Dump streamline data to array
+    streamlines_data = np.vstack(
+        [
+            flipped_streamlines[i][
+                np.newaxis,
+            ]
+            for i in range(len(flipped_streamlines))
+        ]
+    )
+
+    if streamline_class_name == "plausible":
+        streamlines_class = 0
+    elif streamline_class_name == "implausible":
+        streamlines_class = 100
```

Comment:
We'll need to improve this so that it is not hard-coded.

The block may best fit into a dedicated method.